### ISSUE TITLE
Clients aren't respecting tags with multi-monitors (in VMware Workstation)

### ISSUE THREAD

**Issue by PhilomathJ** [kosorin/awesome-rice#2]
# Issue Description

When using triple monitors with this [kosorin/awesome-rice](https://github.com/kosorin/awesome-rice) cloned repo, clients get super buggy to being unusable. The clients do not seem to respect the tags at all, always being visible regardless of tag selected.

This is running in a VM on VMware Workstation. I do not currently have a bare-metal install on which to test this

## Details

When triple monitors are activated: 
- Three independent desktops appear as expected
- Each monitor's tag list appears to work independently as expected
- Switching tags updates the tag list in the top bar for just that monitor as expected
- Client windows can be dragged to each monitor as expected

However, when switching or toggling tags, the clients displayed don't really seem to update (show/hide) when different tag(s) are selected/toggled. The environment quickly becomes unusable with unresponsive clients and overlapping screen artifacts.

### VMware

When VM is only given only one monitor by VMware Workstation, everything works great. When VM is given two or three monitors by VMware Workstation, the behavior above is present

### Default Vanilla AwesomeWM

When replacing this code with a default vanilla install of [AwesomeWM](https://github.com/awesomeWM/awesome) everything works great as expected. Each monitor has its own independent desktop with tag list. Clients drag between monitors and respect the independent tags as expected.

### Planned Tests

1. I would like to see if this behavior is present in a bare-metal install, This will determine if the issue is with the VMware display monitor/screen settings.
    1. See attached `xrandr` outputs for both single and triple monitors  
    [xrandr-single.txt](https://github.com/kosorin/awesome-rice/files/13219429/xrandr-single.txt)
[xrandr-triple.txt](https://github.com/kosorin/awesome-rice/files/13219430/xrandr-triple.txt)

1. I am planning on going back in your commit history and trying out different releases to see where this behavior is introduced.

# Environment

## Host

Intel i7-6700
32GB DDR4
Nvidia GTX 1080Ti
Windows 10.0.19045
VMware Workstation 16.2.5 Pro

## VM

Manjaro 23.0 KDE Plasma booting into AwesomeWM only
4 cores
8GB RAM
Latest pull of [kosorin/awesome-rice #5ebc510](https://github.com/kosorin/awesome-rice/tree/5ebc510f9862fc4042df8269de026badc83764c2)

**Note:** This same behavior was present on my original pull [commit #90bb007 (Sept 17, 2023)](https://github.com/kosorin/awesome-rice/tree/90bb007e28b791cb8cb0011494c64c92b094ea8f)

**Comment by PhilomathJ** [comment]
I just branched and hard reset the repo back to [kosorin/awesome-rice #fcc46d2 (Jan 24, 2023)](https://github.com/kosorin/awesome-rice/tree/fcc46d2dadbdd47204195432da67f20d28c57b0e). 

**I did experience the same client/screen anomalies as in HEAD**

There were a few errors, being that young a repo, but they appeared to mostly be with a weather widget.

At this point, I would really like to hear confirmation from someone on a bare-metal install that a multi-monitor setup works with this rice as intended. My gut says it's a VMware virtual screen issue (see the attached `xrandr` captures in the original comment). That being said though, vanilla [AwesomeWM](https://github.com/awesomeWM/awesome) does indeed work in my same VMware setup perfectly as expected.

**Comment by kosorin** [comment]
I currently use only single monitor. A few months ago I tested a dual monitor setup and it worked fine, but since then I have made many changes. I'll take a look at it.

**Comment by kosorin** [comment]
I tested a setup with three physical monitors and have not experienced any issue. So I assume this is a VM problem. 

Are you sure the clients didn't have the sticky flag on?

<details>
  <summary>xrandr output</summary>

```
Screen 0: minimum 8 x 8, current 7040 x 1440, maximum 32767 x 32767
DVI-D-0 connected 1680x1050+0+390 (normal left inverted right x axis y axis) 434mm x 270mm
   1680x1050     59.88*+  59.95
   1440x900      59.89
   1280x1024     75.02    60.02
   1280x960      60.00
   1152x864      75.00
   1024x768      75.03    60.00
   800x600       75.00    60.32    56.25
   640x480       75.00    59.94
HDMI-0 connected 1920x1080+5120+360 (normal left inverted right x axis y axis) 510mm x 290mm
   1920x1080     60.00*+  59.94    50.00    60.05    60.00    50.04
   1680x1050     59.95
   1280x1024     60.02
   1280x960      60.00
   1280x720      60.00    59.94    50.00
   1152x864      60.00
   1024x768      60.00
   800x600       60.32
   720x576       50.00
   720x480       59.94
   640x480       59.94    59.93
DP-0 connected primary 3440x1440+1680+0 (normal left inverted right x axis y axis) 800mm x 335mm
   3440x1440     59.97*+  49.99    29.99
   2560x1080     60.00    59.94    50.00
   1920x1080     60.00    59.94    50.00
   1680x1050     59.95
   1600x900      60.00
   1280x1024     60.02
   1280x800      59.81
   1280x720      60.00    59.94    50.00
   1152x864      59.96
   1024x768      60.00
   800x600       60.32
   720x576       50.00
   720x480       59.94
   640x480       59.94    59.93
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 disconnected (normal left inverted right x axis y axis)
DP-3 disconnected (normal left inverted right x axis y axis)
DP-4 disconnected (normal left inverted right x axis y axis)
DP-5 disconnected (normal left inverted right x axis y axis)
None-1-1 connected (normal left inverted right x axis y axis)
   1024x768      60.00 +
```

</details>

**Comment by PhilomathJ** [comment]
Thanks for the update. What distro were you using for the triple monitor test? All my testing has been on Manjaro so far.

**Comment by kosorin** [comment]
Arch

**Comment by PhilomathJ** [comment]
I am now 99% sure my issue here is bc VMware is using its own graphics driver (see attached `inxi -Faz` output). VMware Workstation does not have direct hardware access to the Nvidia card, therefor cannot install nonfree or even free drivers. Since the issue I see only presents when three screens are activated, I believe it is a video memory issue.

I did previously test my setup with the default AwesomeWM. That works just fine, but it is also bare-bones with no heavy lifting on the graphics part. This also supports why you see now issue when running on bare-metal.

I see now my options are to either run bare-metal as well, or with a type1 hypervisor (ESXi or Proxmox).

Closing this issue.

[inxi-vmware.md](https://github.com/kosorin/awesome-rice/files/13314331/inxi-vmware.md)

**Comment by PhilomathJ** [comment]
I can now confirm that this issue was solely bc of the way Microsoft and Windows 10 (host) force their crappy Hyper-V shit on by default; even forcing VMware to use it. I disabled one setting on my windows host, effectively turned off forcing all virtualized graphics through Hyper-V. Now everything with [@kosorin](https://github.com/kosorin)'s rice here works as expected (via VMware Workstation 16) even with triple monitors enabled.

I found the answer in this [r/vmware reddit post](https://www.reddit.com/r/vmware/comments/11fsuqp/comment/jap7ti5/?utm_source=share&utm_medium=web2x&context=3)

The specific article referenced is this one from [makeuseof](https://www.makeuseof.com/windows-11-disable-vbs/)

The exact single command from the above article that worked for me is:
(run from an administrative command prompt on my Windows 10 host)
```
 bcdedit /set hypervisorlaunchtype off
```
then reboot.
